### ISSUE TITLE
Using a trained model saved in native pytorch format fails to reproduce train accuracy

### ISSUE THREAD

**Issue by r-kellerm** [intel/openfl#609]
**Describe the bug**
Using [PR#550](https://github.com/intel/openfl/pull/550) and saving best or last model snapshots fails to reproduce the accuracy reported by the federated train environment

**To Reproduce**
1. Modify `test_hello_federation.sh` to run torch_cnn_mnist.
2. Add the following lines at the end of test_hello_federation.sh: `fx model save -i "./save/${TEMPLATE}_best.pbuf" -o $"./save/${TEMPLATE}_best.pth`" , `fx model save -i "./save/${TEMPLATE}_last.pbuf" -o $"./save/${TEMPLATE}_last.pth"`
3. Run `test_hello_federation.sh` till completion, the accuracy of the aggregated model should be >0.98.
4. Run the code in the attached file. The reported accuracy for the same model is now 0.58.

**Expected behavior**
The validation accuracy for a stand alone native torch model should be the same as what is reported from OpenFL.

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.1 LTS
 - PyTorch: 1.8, 
[standalone_mnist_cnn.py.txt](https://github.com/intel/openfl/files/10050560/standalone_mnist_cnn.py.txt)

**Comment by psfoley** [comment]
Thanks @r-kellerm for reporting this. It's possible inconsistent data preprocessing may lead to this sort of issue, but we will take a look at this and get back to you.

**Comment by r-kellerm** [comment]
Thanks @psfoley. The preprocessing is the same - verified it with 2 different models.

**Comment by igor-davidyuk** [comment]
> Thanks @psfoley. The preprocessing is the same - verified it with 2 different models.

Hey @r-kellerm! Thanks for sharing this.

First of all, I must note that OpenFL just provides Federated infrastructure for the user-defined workload. A particular model's accuracy is also reported by user code, so OpenFL core code can hardly affect these matters.

Thank you for attaching the Python script. Looking into it I observed that you did not recreate the data preprocessing pipeline used in the `torch_cnn_mnist` tutorial, as Patrick suggested.
I tried reusing the code from the `src/` folder and compiled [this script](https://github.com/intel/openfl/files/10221427/torch_cnn_script.py.txt). The saved model performed at the same level as was reported by Aggregator.

So please pay attention to the preprocessing you use during training and apply the same manipulations to your test data.

**Comment by r-kellerm** [comment]
Thanks a lot @igor-davidyuk,
I still don't understand what is the missing piece in the no-OpenFL code. What pre-processing step is missing? The shard reference?

Thanks again.

**Comment by igor-davidyuk** [comment]
> Thanks a lot @igor-davidyuk, I still don't understand what is the missing piece in the no-OpenFL code. What pre-processing step is missing? The shard reference?
> 
> Thanks again.

You added an additional preprocessing step, which is `transforms.ToTensor()`. This transform normalizes data, this is why you are getting lower accuracy. 
A quick fix that can solve this is removing this transform or adding `data = 255*data` to your validation loop.
